src/Tasks/RedistList.cs

```diff
@@ -437,6 +448,19 @@ private AssemblyEntry GetUnifiedAssemblyEntry(string assemblyName)
             return unifiedEntry;
         }
 
+        private AssemblyNameExtension GetAssemblyNameExtension(string assemblyName)
+        {
+            AssemblyNameExtension assemblyNameExtension = null;
+
+            if(!_assemblyNameToAssemblyNameExtension.TryGetValue(assemblyName, out assemblyNameExtension))
```

Comment:
You can use `ConcurrentDictionary.GetOrAdd` to make the check and instantiation atomic.